src/WinXP/apps/Notepad/index.js

```diff
@@ -99,24 +162,129 @@ export default function Notepad({ onClose }) {
+
+    if (newSettings) {
+      setFindSettings(newSettings);
+      settings = { ...settings, ...newSettings };
+    }
+    /// Conduct the search
+
+    const index = getIndex(settings);
+    if (index !== -1) selectText(index, index + settings.searchWord.length);
+    else
+      openApp(
+        'InfoDialog',
+        { info: `Cannot find "${settings.searchWord}"` },
+        {
+          header: { ...appSettings.InfoDialog.header, title: 'Notepad' },
+        },
+      );
+  };
+
+  const getIndex = ({ forwardSearch, searchWord, caseSensitive }) => {
+    let searchStr = textareaRef.current.value;
+
+    if (!caseSensitive) {
+      searchStr = docText.toLowerCase();
+      searchWord = searchWord.toLowerCase();
+    }
+
+    /// Reaching index 0 in 'up' search
+    if (!forwardSearch && caretStart.current === 0) return -1;
+
+    return forwardSearch
+      ? searchStr.indexOf(searchWord, caretEnd.current)
+      : searchStr.lastIndexOf(searchWord, caretStart.current - 1);
+  };
+
+  function openApp(app, props, customAppSettings) {
+    appContext.dispatch({
+      type: ADD_APP,
+      payload: {
+        ...appSettings[app],
+        injectProps: props,
+        customAppSettings,
+      },
+    });
+    /// Reselect text
+    selectText(caretStart.current, caretEnd.current);
+  }
+
+  function onOpenFind() {
+    openApp('FindDialog', { findSettings, findNext, updateSettings });
+  }
+
+  function onOpenReplace() {
+    openApp('ReplaceDialog', {
+      findSettings,
+      findNext,
+      onReplace,
+      onReplaceAll,
+      updateSettings,
+    });
+  }
+
+  const frameInitialContent = `
+  <!DOCTYPE html><html>
+  <head>
+  <style>
+  html, div, body, textarea {
+    height: 100%;
+    width: 100%;
+    margin: 0;
+  }
+  html, body {
+    overflow: hidden;
+  }
+  textarea {
+    flex: auto;
+    outline: none;
+    font-family: 'Lucida Console', monospace;
+    font-size: 13px;
+    line-height: 14px;
+    resize: none;
+    padding: 2px;
+    ${wordWrap ? '' : 'white-space: nowrap; overflow-x: scroll;'}
+    overflow-y: scroll;
+    border: 1px solid #96abff;
+  }
+
+  textarea::selection {
+    background-color: #1660e8;
+    color: white;
+  }
+  </style>
+  </head>
+  <body><div></div></body></html>`;
+
   return (
     <Div>
       <section className="np__toolbar">
         <WindowDropDowns items={dropDownData} onClickItem={onClickOptionItem} />
       </section>
-      <StyledTextarea
-        ref={textareaRef}
-        wordWrap={wordWrap}
-        value={docText}
-        onChange={e => setDocText(e.target.value)}
-        onKeyDown={onTextAreaKeyDown}
-        onSelect={e => {
-          const { selectionStart, selectionEnd, value } = e.target;
-          setSelectedText(value.substring(selectionStart, selectionEnd));
-          setCaretPos([selectionStart, selectionEnd]);
-        }}
-        spellCheck={false}
-      />
+
+      <Frame initialContent={frameInitialContent} ref={frameRef} tabIndex={-1}>
```

Comment:
tabIndex is fine with either a string or a number, but consistency is appreciated 🙏